teos/src/carrier.rs

```diff
@@ -139,77 +147,40 @@ impl Carrier {
         receipt
     }
 
-    /// Gets the block height at where a given [Transaction] was confirmed at (if any).
-    fn get_tx_height(&self, txid: &Txid) -> Option<u32> {
-        if let Some(block_hash) = self.get_block_hash_for_tx(txid) {
-            self.get_block_height(&block_hash)
-        } else {
-            None
-        }
-    }
-
-    /// Queries the height of a given [Block](bitcoin::Block). Returns it if the block can be found. Returns [None] otherwise.
-    fn get_block_height(&self, block_hash: &BlockHash) -> Option<u32> {
-        self.hang_until_bitcoind_reachable();
-
-        match self.bitcoin_cli.get_block_header_info(block_hash) {
-            Ok(header_data) => Some(header_data.height as u32),
-            Err(JsonRpcError(RpcError(rpcerr))) => match rpcerr.code {
-                rpc_errors::RPC_INVALID_ADDRESS_OR_KEY => {
-                    log::info!("Block not found: {}", block_hash);
-                    None
-                }
-                e => {
-                    log::error!("Unexpected error code when calling getblockheader: {}", e);
-                    None
-                }
-            },
-            Err(JsonRpcError(TransportError(_))) => {
-                // Connection refused, bitcoind is down.
-                log::error!("Connection lost with bitcoind, retrying request when possible");
-                self.flag_bitcoind_unreachable();
-                self.get_block_height(block_hash)
-            }
-            // TODO: This may need finer catching.
-            Err(e) => {
-                log::error!("Unexpected JSONRPCError when calling getblockheader: {}", e);
-                None
-            }
-        }
-    }
-
-    /// Gets the block hash where a given [Transaction] was confirmed at (if any).
-    pub(crate) fn get_block_hash_for_tx(&self, txid: &Txid) -> Option<BlockHash> {
+    /// Checks whether a given transaction can be found in the mempool.
+    pub(crate) fn in_mempool(&self, txid: &Txid) -> bool {
         self.hang_until_bitcoind_reachable();
 
         match self.bitcoin_cli.get_raw_transaction_info(txid, None) {
```

Comment:
Ahh probably my main point of confusion while reviewing this is I forgot that the behavior of get_raw_transaction_info() changes based on whether txindex is on or off. Not sure if a comment/note here briefly explaining this (when txindex is off, it only looks to see if the tx is in the mempool) would help other reviewers?

But also a question here ~ If a user _is_ using bitcoin core with txindex (for something else maybe), could this lead to unexpected behavior? Like get_raw_transaction_info won't just be doing what in_memool() intended and checking if it's in the mempool, it'll return the tx even if it's confirmed?

Reply:
> Ahh probably my main point of confusion while reviewing this is I forgot that the behavior of get_raw_transaction_info() changes based on whether txindex is on or off. Not sure if a comment/note here briefly explaining this (when txindex is off, it only looks to see if the tx is in the mempool) would help other reviewers?

It may actually, will add that as a comment.

> But also a question here ~ If a user _is_ using bitcoin core with txindex (for something else maybe), could this lead to unexpected behavior? Like get_raw_transaction_info won't just be doing what in_memool() intended and checking if it's in the mempool, it'll return the tx even if it's confirmed?

That is true indeed. I'll patch this so, if the transaction is pulled, then the confirmation count is also checked. I may rename revamp this so it returns either `InMempoolSince(now)` or `ConfirmedIn(x)` based on the response.

Nice catch!

Reply:
Sounds good to me!